File: flashvid-platform-gin/internal/model/topic.go
package model

// 话题信息
type TopicInfo struct {
	ID 		int64  `json:"id"`          // 话题ID
	Name 	string `json:"name"`        // 话题名称
	Description string `json:"description"` // 话题描述
	CoverURL    string `json:"coverUrl"`   // 话题封面
	ViewCount   int64  `json:"viewCount"`   // 浏览量
	VideoCount  int32  `json:"videoCount"`  // 视频数
	CreatedAt   string `json:"createdAt"`   // 创建时间
}

// 话题列表输出
type TopicListOutput struct {
	Topics          []TopicInfo `json:"topics"`          // 话题列表
	NextCursorToken string      `json:"nextCursorToken"` // 下次请求传这个，空字符串表示没有更多
	HasMore         bool        `json:"hasMore"`         // 是否还有更多数据
}

// 根据话题ID获取话题下的视频列表输出
type GetTopicVideosOutput struct {
	Videos          []VideoInfo `json:"videos"`          // 视频列表
	NextCursorToken string      `json:"nextCursorToken"` // 下次请求传这个，空字符串表示没有更多
	HasMore         bool        `json:"hasMore"`         // 是否还有更多数据
}

File: frontend/src/api/topic.ts
import http from './http'
import type { ApiResponse } from '@/types/api'
import type { FeedVideo } from './feed'

export interface TopicItem {
  id: string
  name: string
  description: string
  coverUrl: string
  viewCount: number
  videoCount: number
  createdAt: string
}

export interface TopicsResponse {
  topics: TopicItem[]
  nextCursorToken: string
  hasMore: boolean
}

export interface TopicVideosResponse {
  videos: FeedVideo[]
  nextCursorToken: string
  hasMore: boolean
}

export type TopicSort = 'hot' | 'latest'

export interface GetTopicsParams {
  sort?: TopicSort
  count?: number
  cursor?: string
}

export interface GetTopicVideosParams {
  sort?: 'popular' | 'latest'
  cursor?: string
  count?: number
}

export const getTopics = (params: GetTopicsParams = {}) =>
  http.get<ApiResponse<TopicsResponse>>('/topics', { params })

export const searchTopics = (keyword: string, cursor?: string, count?: number) =>
  http.get<ApiResponse<TopicsResponse>>('/topics/search', { params: { keyword, cursor, count } })

// 话题详情
export const getTopicById = (topicId: string) =>
  http.get<ApiResponse<{ topic: TopicItem }>>(`/topics/${topicId}`)

// 话题下的视频列表（popular / latest）
export const getTopicVideos = (topicId: string, params: GetTopicVideosParams = {}) =>
  http.get<ApiResponse<TopicVideosResponse>>(`/topics/${topicId}/videos`, { params })
